src/components/PeoplesPage/PeoplesPage.tsx

```diff
@@ -0,0 +1,68 @@
+import { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { getPeople } from '../../api';
+import { Person } from '../../types';
+import { findParentByName } from '../../utiles/findParentByName';
+import { Loader } from '../Loader';
+import { PeopleTable } from '../PeoplesTable';
+
+export const PeoplesPage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoaded, setIsLoaded] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const { slug = '' } = useParams();
+
+  const fetchPeople = async () => {
+    setIsLoaded(true);
+    try {
+      const peopleFromServer = await getPeople();
+
+      setPeople(peopleFromServer);
+    } catch {
+      setIsError(true);
+    } finally {
+      setIsLoaded(false);
+    }
+  };
+
+  useEffect(() => {
+    fetchPeople();
+  }, []);
+  const isPeopleFromServer = !isLoaded && people.length === 0;
+
+  const prepearedPeople = people.map(person => {
+    const mother = findParentByName(people, person.motherName);
+    const father = findParentByName(people, person.fatherName);
+
+    return { ...person, mother, father };
+  });
```

Comment:
will be better to move it to separate util function
and transform people array only once after retrieving them from server instead of do this on each component rendering